File: app/src/main/java/com/android/ashwiask/tvmaze/di/AppModule.java
package com.android.ashwiask.tvmaze.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ashwini Kumar.
 */
@Module(includes = ViewModelModule.class)
class AppModule {
    @Provides
    @Singleton
    Context provideContext(Application application) {
        return application;
    }
}

File: app/src/main/java/com/android/ashwiask/tvmaze/di/AppModule.kt
package com.android.ashwiask.tvmaze.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Ashwini Kumar.
 */
@Module(includes = [ViewModelModule::class])
internal class AppModule {
    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }
}
